packages/babel-traverse/src/scope/index.js

```diff
@@ -496,6 +496,11 @@ export default class Scope {
     for (const name in ids) {
       for (const id of (ids[name]: Array<Object>)) {
         let local = this.getOwnBinding(name);
+
+        // Ignore existing binding if it's the name of the current function or class
+        // expression
+        if (local && local.kind === "local") local = null;
```

Comment:
This solves it, but I think the appropriate place to solve it is in `#checkBlockScopedCollisions` so others can use it. https://github.com/motiz88/babel/blob/73cd1ce8d02cd3035593007ff713f7aa7cff978d/packages/babel-traverse/src/scope/index.js#L336-L346